src/target.cr

```diff
@@ -1,21 +1,40 @@
 module Shards
-  class Target < Hash(String, String)
+  class Target
     property name : String
+    property main : String? = nil
+
+    getter attributes : Hash(String, YAML::Any)
 
     def self.new(pull : YAML::PullParser) : self
-      Target.new(pull.read_scalar).tap do |target|
-        pull.each_in_mapping do
-          target[pull.read_scalar] = pull.read_scalar
+      name = pull.read_scalar
+      main = nil
+      extra_attributes = {} of String => YAML::Any
+
+      pull.each_in_mapping do
+        case key = pull.read_scalar
+        when "main"
+          main = pull.read_scalar
+        else
+          extra_attributes[key] = YAML::Any.new(pull.read_scalar)
         end
       end
+
+      unless main
+        raise "Missing attribute 'main' for target #{name.inspect}"
+      end
+
+      Target.new(name, main, extra_attributes)
+    end
+
+    def initialize(@name : String, @main : String, @attributes = {} of String => YAML::Any)
     end
 
-    def initialize(@name)
-      super()
+    def source_path : String
+      main
     end
 
-    def main
-      self["main"]
+    def output_path : String
+      File.join(Shards.bin_path, name)
     end
```

Comment:
Again, this is nice but this is only ever used once, and introduces a dependency on a configuration variable (`Shards.bin_path`) that's external to SPEC.